app/scripts/components/form/dropdown.js

```diff
@@ -21,39 +23,68 @@ function renderMenu (items, value, style) {
   );
 }
 
+/**
+ * Translator for status objects.
+ *
+ * @param {Object} statusObject
+ * @param {string} status - url query parameter to match. e.g. "running", "completed", "failed"
+ * @returns {string} - either the correct label from the status Object, or the input status.
+ */
+const statusToLabel = (statusObject = {}, status) => {
+  let result = Object.keys(statusObject).filter(
+    (label) => statusObject[label] === status
+  );
+  if (result.length) return result[0];
+  return status;
+};
+
 class Dropdown extends React.Component {
-  constructor () {
-    super();
+  constructor (props) {
+    super(props);
     this.displayName = 'Dropdown';
+    const initialValue = statusToLabel(props.options, initialValueFromLocation(props));
     this.state = {
-      value: ''
+      value: initialValue
     };
     this.onSelect = this.onSelect.bind(this);
     this.onChange = this.onChange.bind(this);
   }
 
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    const {location, paramKey, dispatch, action, options} = this.props;
+    if (location.query[paramKey] !== prevProps.location.query[paramKey]) {
+      let value = location.query[paramKey];
+      dispatch(action({ key: paramKey, value }));
+      if (value) value = statusToLabel(options, value);
+      this.setState({value}); // eslint-disable-line react/no-did-update-set-state
```

Comment:
You only want to do `this.setState` if you got a value from `statusToLabel`, correct? So this should this statement be inside the `if` block?

Reply:
No, because we always want to set the state's value no matter what the user has typed, but if the user types running, we will correct it to Running.

And if the user deleted everything in the box, we want to set the value to the empty string.

Reply:
Ah, got it